name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/dnd/SecurityDragListener.java

```diff
@@ -20,34 +24,37 @@ public SecurityDragListener(StructuredViewer viewer)
     @Override
     public void dragSetData(DragSourceEvent event)
     {
-        SecurityTransfer.getTransfer().setSecurity(getSecurity());
+        SecurityTransfer.getTransfer().setSecurities(getSecurities());
     }
 
     @Override
     public void dragStart(DragSourceEvent event)
     {
-        event.doit = getSecurity() != null;
+        event.doit = getSecurities() != null;
     }
 
-    private Security getSecurity()
+    private List<Security> getSecurities()
     {
-        IStructuredSelection selection = (IStructuredSelection) viewer.getSelection();
+        IStructuredSelection selection = (IStructuredSelection) viewer.getStructuredSelection();
         if (selection.isEmpty())
             return null;
-
-        Object element = selection.getFirstElement();
-
-        if (element instanceof Security)
-        {
-            return (Security) element;
-        }
-        else if (element instanceof Adaptable)
-        {
-            return ((Adaptable) element).adapt(Security.class);
-        }
-        else
-        {
-            return null;
+        
+        List<Security> selectedSecurities = new ArrayList<>();
+        Iterator<Security> selectionIterator = selection.iterator();
+        while(selectionIterator.hasNext()) {
+            Object object = selectionIterator.next();
+            Security selectedSecurity;
+            if(object instanceof Security) {
+                selectedSecurity = (Security) object;
+                selectedSecurities.add(selectedSecurity);
+            }else if(object instanceof Adaptable) {
+                selectedSecurity = ((Adaptable) object).adapt(Security.class);
+                selectedSecurities.add(selectedSecurity);
```

Comment:
I believe we need to have a check for *null* here. Previously, it there was only one adoptable, returning null was not a problem because it meant just that: no security selected. However, in the list of securities, there must not be null values.